### PR TITLE
circleCI settings for thiccc branch

### PR DESCRIPTION
**Pull request by tomahawk28**
Toward https://github.com/giantswarm/giantswarm/issues/10798

New circleCI setting for thiccc branch only.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -18,23 +43,23 @@ workflows:
...
           password_envar: "QUAY_PASSWORD"
           requires:
-            - go-build
-          filters:
-            # Trigger the job also on git tag.
-            tags:
-              only: /^v.*/
+            - build
 
       - architect/push-to-docker:
```
Comment (rossf7) [inline .circleci/config.yml:48]: I think we need to use push-to-docker-legacy so the image tag is correct for draughtsman.

See https://github.com/giantswarm/aws-operator/blob/ca543815a0672225ca14396649ce51f9fe7502d2/.circleci/config.yml#L258

**Comment by rossf7** [comment]
For draughtsman the tag is just the SHA without a version prefix. If you check the build step you can see it pushes the image to Quay.

https://circleci.com/gh/giantswarm/app-operator/4647

So we don't need the `push-app-operator-to-quay` job. But it doesn't push to Aliyun. So I think we need push-to-docker-legacy for that.

**Comment by tomahawk28** [comment]
I think we need to revert this in installations, right? https://github.com/giantswarm/installations/pull/1181

**Comment by rossf7** [comment]
> I think we need to revert this in installations, right?

Yes but there is also the project list in architect. I would check the flattening docs to make sure we don't miss anywhere.

**Pull request merged**